src/Symfony/Component/Mailer/Transport.php

```diff
@@ -169,6 +169,12 @@ private static function createTransport(string $dsn, EventDispatcherInterface $d
                 }
 
                 throw new LogicException(sprintf('The "%s" scheme is not supported for mailer "%s".', $parsedDsn['scheme'], $parsedDsn['host']));
+            case 'custom':
+                if (!$customTransportClass) {
+                    throw new \LogicException('You must specify the transport class when using the "custom" provider.');
+                }
+
+                return new $customTransportClass($user, $pass, $query, $client, $dispatcher, $logger);
```

Comment:
The `TransportInterface` does not include the class constructor, so you don't know here how to build the custom constructor class.

Also you'd have to check if `$customTrasportClass` implements `TransportInterface`

Reply:
Even if the interface does not include the constructor, we can pass the more arguments we have, to enable the developer to do anything he needs. Also, the developer can implements the TransportInterface or extends one of the Abstract*Transport classes.

I have added the class implements check